File: mobileapp/lib/models/event.dart
class Event {
  final String id;
  final String title;
  final String description;
  final DateTime eventDate;
  final String venue;
  final String? imageUrl;
  final bool hasReminder;
  final DateTime createdAt;

  Event({
    required this.id,
    required this.title,
    required this.description,
    required this.eventDate,
    required this.venue,
    this.imageUrl,
    this.hasReminder = false,
    required this.createdAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['_id'] ?? '',
      title: json['title'] ?? '',
      description: json['description'] ?? '',
      eventDate: DateTime.parse(json['eventDate'] ?? DateTime.now().toIso8601String()),
      venue: json['venue'] ?? '',
      imageUrl: json['imageUrl'],
      hasReminder: json['hasReminder'] ?? false,
      createdAt: DateTime.parse(json['createdAt'] ?? DateTime.now().toIso8601String()),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'title': title,
      'description': description,
      'eventDate': eventDate.toIso8601String(),
      'venue': venue,
      'imageUrl': imageUrl,
      'hasReminder': hasReminder,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: mobileapp/backend/src/models/Event.js
const mongoose = require('mongoose');

const eventSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: [true, 'Title is required'],
      trim: true,
    },
    description: {
      type: String,
      required: [true, 'Description is required'],
    },
    eventDate: {
      type: Date,
      required: [true, 'Event date is required'],
    },
    startTime: {
      type: String,
      required: [true, 'Start time is required'],
    },
    endTime: {
      type: String,
      required: [true, 'End time is required'],
    },
    venue: {
      type: String,
      required: [true, 'Venue is required'],
    },
    category: {
      type: String,
      enum: ['Academic', 'Sports', 'Cultural', 'Workshop', 'Seminar', 'Other'],
      default: 'Other',
    },
    imageUrl: {
      type: String,
      default: null,
    },
    organizer: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    registeredUsers: [
      {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
      },
    ],
    capacity: {
      type: Number,
      default: null,
    },
    isFeatured: {
      type: Boolean,
      default: false,
    },
    createdAt: {
      type: Date,
      default: Date.now,
    },
    updatedAt: {
      type: Date,
      default: Date.now,
    },
  },
  { collection: 'events' }
);

eventSchema.pre(/^find/, function (next) {
  this.populate({
    path: 'organizer',
    select: 'name email department',
  }).populate({
    path: 'registeredUsers',
    select: 'name email',
  });
  next();
});

eventSchema.pre('save', function (next) {
  this.updatedAt = Date.now();
  next();
});

module.exports = mongoose.model('Event', eventSchema);
